newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,375 @@
+
+  const renderImage = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): React.Node => (
+      <img
+        src={item.thumbnailUrl}
+        style={{
+          ...styles.image,
+          height: imageHeight,
+          minHeight: imageHeight,
+          width: cellWidth,
+        }}
+        alt={item.title}
+        title={item.title}
+      />
+    ),
+    [cellWidth, imageHeight]
+  );
+
+  const renderThumbnail = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!item.skeleton && !item.link && !item.thumbnailUrl) return null;
+      if (item.link) {
+        return (
+          <a
+            href={item.link}
+            target="_blank"
+            onFocus={onFocusItem}
+            className={classesForLink.link}
+          >
+            {renderImage(item)}
+          </a>
+        );
+      }
+      if (item.thumbnailUrl) {
+        return renderImage(item);
+      }
+      if (item.skeleton) {
+        return (
+          <Skeleton variant="rect" height={imageHeight} width={cellWidth} />
+        );
+      }
+    },
+    [renderImage, cellWidth, imageHeight]
+  );
+
+  const renderItemTitle = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!displayItemTitles) return null;
+      return (
+        <>
+          <Spacer />
+          {item.title ? (
+            <div style={{ ...styles.itemTitleContainer, width: cellWidth }}>
+              <Text noMargin style={styles.itemTitle}>
+                {item.title}
+              </Text>
+            </div>
+          ) : (
+            <Skeleton
+              variant="rect"
+              height={titleHeight}
+              width={(cellWidth / 3) * (1 + 2 * Math.random())} // Make rectangles of different lengths so that the UI feels more "alive".
+            />
+          )}
+        </>
+      );
+    },
+    [cellWidth, titleHeight]
+  );
+
+  const computeScroll = React.useCallback(
+    (
+      direction: 'left' | 'right',
+      scrollViewElement: HTMLUListElement
+    ): number => {
+      const visibleThumbnailsCount = Math.floor(
+        scrollViewElement.offsetWidth / widthUnit
+      );
+      const scale = visibleThumbnailsCount * widthUnit;
+
+      const currentScroll = scrollViewElement.scrollLeft;
+      const currentFirstVisibleItemIndex = currentScroll / widthUnit;
+
+      if (
+        direction === 'right' &&
+        currentFirstVisibleItemIndex >
+          itemsToDisplay.length - visibleThumbnailsCount - 1
+      )
+        return 0;
+      return roundScroll(
+        scrollViewElement.scrollLeft + scale * (direction === 'left' ? -1 : 1)
+      );
+    },
+    [widthUnit, itemsToDisplay]
+  );
+
+  const roundScroll = React.useCallback(
+    (value: number): number => {
+      return Math.round(value / widthUnit) * widthUnit;
+    },
+    [widthUnit]
+  );
+
+  const onClickArrow = (direction: 'left' | 'right') =>
```

Comment:
On a very small viewport (use the Storybook viewport "small mobile"), the scroll won't scroll because I guess it's smaller than one tile. There is maybe a Math.min to add somewhere to ensure we scroll at least the size of a tile or at least to the next "logical" page:

![image](https://user-images.githubusercontent.com/1280130/144216143-0b7bc1f6-f078-4ef3-91f4-d851b83c01c9.png)